api/pkg/authz/enforcer/cache.go

```diff
@@ -0,0 +1,102 @@
+package enforcer
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/ory/keto-client-go/models"
+	cache "github.com/patrickmn/go-cache"
+)
+
+type InMemoryCache struct {
+	store *cache.Cache
+
+	mapLock sync.Mutex
+
+	// Instead of caching the full names of the subject, resource and action, ids will be
+	// generated (using an incrementing counter) for each unique value. This will aid in
+	// generating smaller cache keys.
+	// The following maps store the mapping of the name -> internal id.
```

Comment:
Curious, is it possible to avoid storing the mapping altogether and use some hashing function with a low collision rate instead?

Reply:
I did briefly explore this. We can choose something like SHA-1 and work with the first 7 characters like with git commits. With the authz data we will handle, it's extremely unlikely it will result in any collision. And it's definitely cleaner.

But I figured the lookup map with the auto-incrementing id was just safer.